pkg/interface/src/logic/reducers/contact-update.ts

```diff
@@ -5,25 +5,29 @@ import { ContactUpdate } from '@urbit/api';
 
 import useContactState, { ContactState } from '../state/contact';
 import { reduceState } from '../state/base';
+import { SubscriptionRequestInterface, UrbitInterface } from '@urbit/http-api';
+import { handleSubscriptionError, handleSubscriptionQuit } from '../lib/subscriptionHandlers';
 
 
-export const ContactReducer = (json) => {
+const ContactReducer = (json) => {
   const data: ContactUpdate = _.get(json, 'contact-update', false);
   if (data) {
-    reduceState<ContactState, ContactUpdate>(useContactState, data, [
-      initial,
-      add,
-      remove,
-      edit,
-      setPublic
-    ]);
+    useContactState.getState().set(state => {
+      state = reduceState<ContactState, ContactUpdate>(useContactState, data, [
```

Comment:
!! please don't assign to immer generated variables, this breaks it

Reply:
Yep, these are all marked as merge conflicts anyway. Will use new pattern